### PR TITLE
Remove libgsystem, ostree: update to 2024.5, rpm-ostree: update to 2024.4

### PR DESCRIPTION
**Pull request by BettyRain**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [X] The toolchain has been rebuilt successfully (or no changes were made to it)
- [X] The toolchain/worker package manifests are up-to-date
- [X] Any updated packages successfully build (or no packages were changed)
- [X] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [X] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [X] All package sources are available
- [X] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [X] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [X] All source files have up-to-date hashes in the `*.signatures.json` files
- [X] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [X] Documentation has been updated to match any changes to the build system
- [X] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?
Remove libgsystem, ostree: update to 2024.5, rpm-ostree: update to 2024.4

###### Change Log  

- Update ostree to 2024.5
- Update rpm-ostree to 2024.4
- Remove libgsystem since the package was deprecated a while ago

###### Does this affect the toolchain?  

**NO**

###### Test Methodology

- Local build
- Full build id: [549128](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=549128&view=results)
- Buddy build id: [549604](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=549604&view=results)

### CODE REVIEW

File: SPECS/ostree/ostree.spec [hunk 1]
```diff
@@ -123,16 +129,16 @@ install -vdm 755 %{buildroot}%{_sysconfdir}/ostree/remotes.d
 %{_unitdir}/ostree*.service
 %{_unitdir}/ostree-finalize-staged.path
 %{_libdir}/dracut/modules.d/98ostree/*
-%{_mandir}/man1/ostree-admin*
+%{_mandir}/man*/*.gz
```
Comment (sameluch) [inline SPECS/ostree/ostree.spec:132]: Should we be slightly more specific here? This seems like a really broad wildcard. Does this claim ownership of more files than it should?

File: SPECS/ostree/ostree.spec [hunk 2]
```diff
@@ -123,16 +129,16 @@ install -vdm 755 %{buildroot}%{_sysconfdir}/ostree/remotes.d
...
-%{_libdir}/systemd/system-preset/91-ostree.preset
 %exclude %{_sysconfdir}/grub.d/*ostree
 %exclude %{_libexecdir}/libostree/grub2*
 %{_libdir}/ostree/ostree-prepare-root
-%{_sysconfdir}/dracut.conf.d/ostree.conf
 %{_libdir}/ostree/ostree-remount
 %{_libdir}/tmpfiles.d/ostree-tmpfiles.conf
-%{_libexecdir}/libostree/*
+%{_libexecdir}/lib%{name}/*
+%{_libexecdir}/installed-tests/*
```
Comment (sameluch) [inline SPECS/ostree/ostree.spec:140]: same here, this seems a bit broad, does ostree make a sub folder in /etc/libexec/installed-tests/?

Comment (BettyRain) [reply SPECS/ostree/ostree.spec:140]: Yeah, and there are about 20+ test files going inside it, so it made sense to make a broader folder.

Comment (BettyRain) [reply SPECS/ostree/ostree.spec:140]: I did make it a bit deeper, tested, it worked!

**Comment by BettyRain** [comment]
Running new build after addressing comments:
[550893](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=550893&view=results)

**Review by sameluch — APPROVED** [review]

**Pull request merged**